frontend/src/components/auth/ResetPassword/SetNewPassword.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button } from "@chakra-ui/react";
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import {
+  ADMIN_SIGNUP_IMAGE,
+  TEACHER_SIGNUP_IMAGE,
+} from "../../../assets/images";
+import { ADMIN_LOGIN, TEACHER_LOGIN } from "../../../constants/Routes";
+import { Role } from "../../../types/AuthTypes";
+import AuthWrapper from "../AuthWrapper";
+import PasswordForm from "../Password/PasswordForm";
+
+const SetNewPassword = ({
+  role,
+  oobCode,
+  email,
+}: {
+  role: Role;
+  oobCode: string;
+  email: string;
+}): React.ReactElement => {
+  const [step, setStep] = useState(1);
+  const history = useHistory();
+  const image = role === "Admin" ? ADMIN_SIGNUP_IMAGE : TEACHER_SIGNUP_IMAGE;
+
+  if (step === 1)
+    return (
+      <AuthWrapper
+        title="Set New Password"
+        subtitle="Please ensure that your new password is different than your old one"
+        image={image}
+        form={
+          <PasswordForm
+            version="ResetPassword"
+            userRole={role}
+            email={email}
+            oobCode={oobCode}
+            setStep={setStep}
+          />
+        }
+      />
+    );
+  return (
+    <AuthWrapper
+      title="Password Reset Successful"
+      subtitle={`Your password has been successfully reset\nClick below to log in`}
+      image={image}
+      form={
+        <Button
+          variant="primary"
+          width="100%"
+          onClick={() =>
+            history.push(role === "Admin" ? ADMIN_LOGIN : TEACHER_LOGIN)
+          }
+        >
+          Login
+        </Button>
+      }
+    />
+  );
+};
```

Comment:
again, I think rather than using `step` here, splitting these into separate components may be a more "React" way of doing things

Reply:
hmm the `step` here is needed to handle going back and forth between components - similar to the logic in TeacherSignup [here](https://github.com/uwblueprint/jump-math/blob/staging/frontend/src/components/auth/TeacherSignup/TeacherSignup.tsx#L42). i figured since there's only 2 steps and the components are just AuthWrapper its unnecessary to have two extra files for them. but lmk if you think otherwise